### PR TITLE
Add VK_ANDROID_frame_boundary for HWUI

### PR DESCRIPTION
**Pull request by hevrard**
VK_ANDROID_frame_boundary is a bespoke extension designed for the
needs of HWUI: add a manual frame boundary to workloads that do not
use vkQueuePresentKHR, but still render frames in sequence, one at a
time (no overlapping frames).

See the description in
gapis/api/vulkan/extensions/android_frame_boundary.api

Bug: b/156105650
Test: manual, on a modified version of our sample

### CODE REVIEW

**Comment by hevrard** [comment]
The modified sample used to test is available with https://github.com/google/agi/pull/672

File: gapis/api/vulkan/extensions/android_frame_boundary.api [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+//
+// Make sure to load this function only when the device supports
+// "VK_ANDROID_frame_boundary", this support will be present only when AGI's
+// spy vulkan layer is present (i.e., when AGI traces the app).
+
+/////////////
+// Structs //
+/////////////
+
+// In due time, this will be declared in the implementation of
```
Comment (pmuetschard) [inline gapis/api/vulkan/extensions/android_frame_boundary.api:34]: Add `TODO` to this comment.

File: gapis/api/vulkan/extensions/android_frame_boundary.api [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+
+//////////////
+// Commands //
+//////////////
+
+@extension("VK_ANDROID_frame_boundary")
+@indirect("VkDevice")
+@override
+@frame_end
+cmd void vkFrameBoundaryANDROID(
```
Comment (pmuetschard) [inline gapis/api/vulkan/extensions/android_frame_boundary.api:49]: Add `@platform("VK_USE_PLATFORM_ANDROID_KHR")` as we only want this on Android.

File: gapis/api/vulkan/templates/vulkan_layer.tmpl [hunk 3]
```diff
@@ -72,6 +72,10 @@
 #include "vulkan/vulkan.h"
 #include "core/vulkan/layer_helpers/vulkan_layer_helpers.h"
 
+// Support VK_ANDROID_frame_boundary, which is not defined by Khronos
+struct AHardwareBuffer;
+typedef void (VKAPI_PTR *PFN_vkFrameBoundaryANDROID)(VkDevice device, VkSemaphore semaphore, VkImage image, AHardwareBuffer* buffer);
+
```
Comment (pmuetschard) [inline gapis/api/vulkan/templates/vulkan_layer.tmpl:78]: Should this be guarded with an Android ifdef?

**Review by pmuetschard — APPROVED** [review]

File: gapis/api/vulkan/extensions/android_frame_boundary.api (hunk 1 above)
Comment (hevrard) [reply gapis/api/vulkan/extensions/android_frame_boundary.api]: done

File: gapis/api/vulkan/extensions/android_frame_boundary.api (hunk 2 above)
Comment (hevrard) [reply gapis/api/vulkan/extensions/android_frame_boundary.api:50]: done

File: gapis/api/vulkan/templates/vulkan_layer.tmpl (hunk 3 above)
Comment (hevrard) [reply gapis/api/vulkan/templates/vulkan_layer.tmpl]: Yes! Done.

**Pull request merged**